modules/uncertainty/R/ensemble.R

```diff
@@ -213,6 +213,28 @@ get.ensemble.samples <- function(ensemble.size, pft.samples, env.samples,
 write.ensemble.configs <- function(defaults, ensemble.samples, settings, model, 
                                    clean = FALSE, write.to.db = TRUE, restart = NULL, rename = FALSE) {
   
+  
+  # --- START OF YOUR NEW CODE ---
+  # Check if there are NO inputs
+  if (is.null(defaults$inputs)) {
+    stop(PEcAn.logger::logger.severe("No inputs provided - cannot configure ensemble"))
+  }
+  
+  # Check each input type (e.g., soil, veg)
+  for (input_type in names(defaults$inputs)) {
+    input_paths <- defaults$inputs[[input_type]]$path
+    
+    # Case: Multiple inputs + no samples → Error
+    if (length(input_paths) > 1 && is.null(ensemble.samples)) {
+      stop(PEcAn.logger::logger.severe(paste(
```

Comment:
logger.severe calls stop for you -- no need for the wrapper.
```suggestion
      PEcAn.logger::logger.severe(paste(
```

Reply:
I also think its worth considering making these logger.error instead of logger.severe. That way users could more easily toggle if they want to workflow to stop here or not. PEcAn is designed to do large ensembles of runs (e.g. Dongchen is currently doing 640000 run ensembles) and you don't necessarily want the entire workflow to come to a halt if there's something off in one of those 640000 runs, which you might not encounter until you're a week into the compute.